### PR TITLE
Add support for `google_compute_node_group` to TGC

### PR DESCRIPTION
**Pull request by delimaneto**
adding support for compute.googleapis.com/NodeGroup
```release-note:none
```

### CODE REVIEW

**Review by melinath — CHANGES_REQUESTED** [review]
Since this is a TGC-only change it should use release note type `none` and the content of the release note should be empty. Release notes are only used for provider changes.

**Comment by sushant00** [comment]
@melinath I think we would need to merge https://github.com/GoogleCloudPlatform/magic-modules/pull/10602 first for NodeTemplate before we can merge this PR for NodeGroup.

Can you please confirm?

**Review by melinath — CHANGES_REQUESTED** [review]
TGC converters shouldn't technically need to depend on each other (in most cases) but if there's a separate PR for node template then we should remove that from here for sure, and we could merge them in order if you prefer.

**Comment by delimaneto** [comment]
> TGC converters shouldn't technically need to depend on each other (in most cases) but if there's a separate PR for node template then we should remove that from here for sure, and we could merge them in order if you prefer.

I just did let NodeTemplate here because is a resource dependency for NodeGroup resource. I think we should merge the other PR about NodeTemplate and after merge here. Make sense?

**Review by melinath — COMMENTED** [review]
sounds good, removing from my queue for now - let me know when node template is done & I'll review this at that point.

**Comment by delimaneto** [comment]
node template is done, thanks.

File: mmv1/templates/tgc/resource_converters.go.erb
Comment (melinath) [inline mmv1/templates/tgc/resource_converters.go.erb:96]:
```suggestion
```

File: mmv1/templates/tgc/resource_converters.go.erb
Comment (melinath) [inline mmv1/templates/tgc/resource_converters.go.erb:83]:
```suggestion
		"google_secret_manager_secret_version": 		   {secretmanager.ResourceConverterSecretManagerSecretVersion()},
```

**Review by melinath — CHANGES_REQUESTED** [review]

File: mmv1/templates/tgc/resource_converters.go.erb
Comment (melinath) [inline mmv1/templates/tgc/resource_converters.go.erb:96]: This is still adding node_template. Please remove this line.

**Review by melinath — CHANGES_REQUESTED** [review]
Please undo the whitespace changes in resource_converters (or do them in a separate PR) - that's going to make this keep running into merge conflicts.

**Comment by melinath** [comment]
Also please resolve the merge conflicts

File: mmv1/templates/tgc/resource_converters.go.erb
Comment (delimaneto) [reply mmv1/templates/tgc/resource_converters.go.erb:96]: if I remove google_compute_node_template, it will not test node template resource anymore. I can't remove it.

Comment (melinath) [reply mmv1/templates/tgc/resource_converters.go.erb:98]: google_compute_node_template support was added in https://github.com/GoogleCloudPlatform/magic-modules/pull/10602, so it doesn't also need to be added here.

**Review by melinath — CHANGES_REQUESTED** [review]

File: mmv1/templates/tgc/resource_converters.go.erb
Comment (delimaneto) [reply mmv1/templates/tgc/resource_converters.go.erb]: ok, done. Thanks

File: mmv1/templates/tgc/resource_converters.go.erb
Comment (melinath) [inline mmv1/templates/tgc/resource_converters.go.erb:49]: please undo these stray changes.

**Review by melinath — CHANGES_REQUESTED** [review]

File: mmv1/templates/tgc/resource_converters.go.erb
Comment (delimaneto) [reply mmv1/templates/tgc/resource_converters.go.erb]: done and aligned lines correctly

**Review by melinath — CHANGES_REQUESTED** [review]
In general, if you're making a bunch of changes to the same file (for example, to fix formatting or whitespace issues), you should make the changes in a separate PR either before you start or after you finish to avoid causing merge conflicts.

I went ahead and merged your other PR that made whitespace changes so that they're done for now. However, that means this PR now has merge conflicts that need to be resolved. But moving forward this will hopefully not be a problem again.

**Review by melinath — CHANGES_REQUESTED** [review]
why are there more whitespace changes? Also, in addition to explaining why, please remove them.

**Comment by delimaneto** [comment]
> why are there more whitespace changes? Also, in addition to explaining why, please remove them.

There are more whitespace changes because I'm removing those on edit file github. I'm resolving the conflicts and sometimes it comes from main branch and I need to fix all of them.

**Review by melinath — CHANGES_REQUESTED** [review]
Please undo the whitespace changes

**Comment by delimaneto** [comment]
> Please undo the whitespace changes

done, thanks

**Review by melinath — CHANGES_REQUESTED** [review]
Taking a look at this, it's only exercising a few of the fields on node_group. Could you add a more complete set of fields? As many as possible, ideally. https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_node_group lists the available fields.

**Comment by delimaneto** [comment]
> Taking a look at this, it's only exercising a few of the fields on node_group. Could you add a more complete set of fields? As many as possible, ideally. https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_node_group lists the available fields.

@melinath we are using the most simple group of fields that resource can once those are just to test the specific resource. As much more simpler, it'll be better as internal team said.

**Review by melinath — CHANGES_REQUESTED** [review]
the only other resource that could be referenced from a node group is a project, which doesn't seem like an excessive thing to add to this configuration. Please add all fields to the test. Thanks!

**Comment by delimaneto** [comment]
> the only other resource that could be referenced from a node group is a project, which doesn't seem like an excessive thing to add to this configuration. Please add all fields to the test. Thanks!

added more fields, thanks.

**Review by sushant00 — APPROVED** [review]

**Comment by delimaneto** [comment]
@melinath could you please review again?

**Review by melinath — APPROVED** [review]
LGTM assuming tests pass

**Comment by delimaneto** [comment]
@melinath could you please make the merge?

**Comment by melinath** [comment]
yep, thanks for the ping!

**Pull request merged**
